add set of authorization plugins

vladak opened the pull request:
This is a set of authorization plugins written by @tulinkry.

tulinkry commented on plugins/LdapPlugin/src/opengrok/auth/plugin/LdapFilter.java:
```diff
@@ -0,0 +1,153 @@
+     * </ul>
+     *
+     * Use \% for printing the '%̈́' character.
+     *
```
This doc actualy lacks info that it replaces any other ldap attribute which would not be ambiguous (lines 119-126)

vladak replied:
you can add it in later PR

vladak replied:
or just send me the diff and I will slap on the top

vladak commented:
Also, the NetBeans projects are probably not working out of the box, can fix them later.

tulinkry commented:
Just to be sure, so it eventually builds the plugins into a jar file and runs the tests or?

tulinkry commented:
And why is junit needed? I thought we already have it since we run some tests

vladak commented:
The tests of the plugins are actually not run. I've added the junit csets as a basis for running the tests via `ant` - not directly related to the auth plugins yet.

vladak commented:
One more note w.r.t. junit - on Travis the tests are run via Maven which I guess knows where to get Junit.jar from. For `ant` though, you had to download the junit.jar yourself and place it in the `lib` directory. It's sort of surprising that I got to fix this only now since it was a nuisance for me from the very start.

tulinkry commented:
ah right, but doing that from netbeans downloads the dependencies automaticaly when you add the library in the test dependencies, right?

vladak commented:
yep, in NB this is easily fixable.

tulinkry commented:
I can't see that maybe I'm just blind but did you delete the file "plugins.jar" from testdata? It seems to me that it is not used anymore.

vladak commented:
It was an attempt to fix #1859 which failed, reverted and renamed test plugins to `testplugins.jar` so they are not confused with `plugins.jar` which actually gets shipped.

vladak commented:
@tulinkry mentioned on Slack that he has not further comments, merging.
